crates/hyperswitch_connectors/src/connectors/novalnet/transformers.rs

```diff
@@ -125,99 +339,712 @@ impl<F, T> TryFrom<ResponseRouterData<F, NovalnetPaymentsResponse, T, PaymentsRe
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: &NovalnetRouterData<&PaymentsCaptureRouterData>,
+    ) -> Result<Self, Self::Error> {
+        let capture_type = if item.router_data.request.is_multiple_capture() {
+            CaptureType::PARTIAL
+        } else {
+            CaptureType::FINAL
+        };
+
+        let reference = item
+            .router_data
+            .request
+            .multiple_capture_data
+            .as_ref()
+            .map(|multiple_capture_data| multiple_capture_data.capture_reference.clone());
+
+        let capture: Option<Capture> = reference.map(|reference| Capture {
+            _type: capture_type,
+            reference,
+        });
+
+        let transaction = NovalnetTransaction {
+            tid: item.router_data.request.connector_transaction_id.clone(),
+            capture,
+            amount: Some(item.amount.to_owned()),
+        };
+
+        let custom = NovalnetCustom {
+            lang: item
+                .router_data
+                .request
+                .get_browser_info()?
+                .get_language()?,
+        };
         Ok(Self {
-            status: common_enums::AttemptStatus::from(item.response.status),
-            response: Ok(PaymentsResponseData::TransactionResponse {
-                resource_id: ResponseId::ConnectorTransactionId(item.response.id),
-                redirection_data: None,
-                mandate_reference: None,
-                connector_metadata: None,
-                network_txn_id: None,
-                connector_response_reference_id: None,
-                incremental_authorization_allowed: None,
-                charge_id: None,
-            }),
-            ..item.data
+            transaction,
+            custom,
         })
     }
 }
 
-//TODO: Fill the struct with respective fields
-// REFUND :
 // Type definition for RefundRequest
+#[derive(Default, Debug, Serialize)]
+pub struct NovalnetRefundTransaction {
+    tid: String,
+    amount: Option<StringMinorUnit>,
+}
+
 #[derive(Default, Debug, Serialize)]
 pub struct NovalnetRefundRequest {
-    pub amount: StringMinorUnit,
+    pub transaction: NovalnetRefundTransaction,
+    pub custom: NovalnetCustom,
 }
 
 impl<F> TryFrom<&NovalnetRouterData<&RefundsRouterData<F>>> for NovalnetRefundRequest {
     type Error = error_stack::Report<errors::ConnectorError>;
     fn try_from(item: &NovalnetRouterData<&RefundsRouterData<F>>) -> Result<Self, Self::Error> {
+        let transaction = NovalnetRefundTransaction {
+            tid: item.router_data.request.connector_transaction_id.clone(),
+            amount: Some(item.amount.to_owned()),
+        };
+
+        let custom = NovalnetCustom {
+            lang: item
+                .router_data
+                .request
+                .get_browser_info()?
+                .get_language()?,
+        };
```

Comment:
Can we pass default language as English if `browser_info` or `language` is not present?